patches.go

```diff
@@ -101,19 +59,12 @@ func ResolveTransforms(ts []v1beta1.Transform, input any) (any, error) {
 // on the "from" resource. Values may be transformed if any are defined on
 // the patch.
 func ApplyFromFieldPathPatch(p PatchInterface, from, to runtime.Object) error {
-	if p.GetFromFieldPath() == "" {
-		return errors.Errorf(errFmtRequiredField, "FromFieldPath", p.GetType())
-	}
```

Comment:
Not strictly related to this PR, but I noticed this was already being handled by `ValidateResources`, which does an even more thorough validation. I figured we could trust that and avoid this redundant layer of validation here.